lighthouse-core/gather/gather-runner.js

```diff
@@ -119,7 +130,16 @@ class GatherRunner {
     await driver.registerPerformanceObserver();
     await driver.dismissJavaScriptDialogs();
     await driver.registerRequestIdleCallbackWrap(options.settings);
-    if (resetStorage) await driver.clearDataForOrigin(options.requestedUrl);
+    if (resetStorage) {
+      const locations = await driver.getImportantLocationsNotCleared(options.requestedUrl);
+      if (locations.length) {
+        LighthouseRunWarnings.push(str_(
+          UIStrings.warningData,
+          {locations: locations.join(', '), locationCount: locations.length}
+        ));
+      }
```

Comment:
wdyt about wrapping this up into the driver function (`driver.getImportantStorageWarning(url, runWarnings)` or whatever)? I don't imagine we're going to be using `getImportantLocationsNotCleared` in other ways, so it makes some sense to keep the warning generation code all together

Reply:
I didn't move all this into the driver function originally because the driver code does not have `UIString`.

I didn't move the driver code into this function because that allows `driver.getImportantLocationsNotCleared()` to be replaced with a dummy function for many existing unit tests. If I do move the driver code in here, I would need to define `sendCommand` for `fakeDriver` because `sendCommand` would be called from a bunch of existing unit tests.  Is there some workaround I could use?

Reply:
Yeah, it's probably arguable if this code makes more sense in `driver` than in `gather-runner`. It's not agnostic driving, and the smarts for _how_ to drive should generally live more in `gather-runner`, but OTOH we're not agnostic about many things in `driver`, including which types of storage to clear or not. Until Fraggle Rock starts changing things, though, personally I think it does make sense in `driver` for now, as part of a pair with `clearDataForOrigin`.

For UIStrings: now that #10630 and `IcuMessage` make it easier to carry localized strings over from the gathering stage, we'll probably start to have more driver/gathering `UIStrings`, so it seems ok to start a new one in this PR.